packages/wrangler/src/zones.ts

```diff
@@ -131,17 +152,41 @@ async function getZoneIdFromHost(from: {
 	const hostPieces = from.host.split(".");
 
 	while (hostPieces.length > 1) {
-		const zones = await fetchListResult<{ id: string }>(
-			`/zones`,
-			{},
-			new URLSearchParams({
-				name: hostPieces.join("."),
-				"account.id": from.accountId,
-			})
-		);
-		if (zones.length > 0) {
-			return zones[0].id;
+		const cacheKey =
+			hostPieces.join(".") +
+			from.accountId +
+			(process.env.VITEST ? process.env.VITEST_CACHE_BUST ?? "" : "");
```

Comment:
I'm really not keen on this approach, even though it is just apparent in testing.
I think I'd rather we create the cache (or cache key) in the code that makes all these requests and pass it down so that it is clear where the state is being used.

Reply:
For example, `process.env` itself is a global so is there not a chance of conflict between parallel tests already?

Reply:
Agreed - I like the approach you took of passing in a map!